### PR TITLE
Migrate from dask[complete] to dask[array] to reduce number of dependencies

### LINKED ISSUE #465: Xclim broken with dask@2.17.1
* xclim version: 0.17.2-beta

### Description

With the newest patch of Dask (2.17.1), the [complete] recipe has been removed. This is causing all builds to fail. We are currently waiting on advice for how to recreate the [complete] installation. For the moment, Xarray can be considered broken with dask@2.17.1.

For more information: https://github.com/dask/dask/pull/6250

### What I Did

```
tox py36
```

### What I Received

```
Testing started at 12:03 p.m. ...
/home/tjs/miniconda3/envs/xclim/bin/python /snap/pycharm-community/194/plugins/python-ce/helpers/pycharm/_jb_tox_runner.py "-e py36"
GLOB sdist-make: /home/tjs/git/xclim/setup.py
py36 create: /home/tjs/git/xclim/.tox/py36
py36 installdeps: coveralls, pytest-cov, pip, setuptools
py36 inst: /home/tjs/git/xclim/.tox/.tmp/package/1/xclim-0.17.2b0.zip
py36 installed: appdirs==1.4.4,attrs==19.3.0,bleach==3.1.5,boltons==20.1.0,Bottleneck==1.3.2,bump2version==1.0.0,certifi==2020.4.5.1,cffi==1.14.0,cfgv==3.1.0,cftime==1.1.3,chardet==3.0.4,click==7.1.2,click-plugins==1.1.1,cligj==0.5.0,coverage==5.1,coveralls==2.0.0,cryptography==2.9.2,dask==2.17.1,distlib==0.3.0,docopt==0.6.2,docutils==0.16,filelock==3.0.12,Fiona==1.8.13.post1,flake8==3.8.2,geojson==2.5.0,geopandas==0.7.0,identify==1.4.17,idna==2.9,importlib-metadata==1.6.0,importlib-resources==1.5.0,jeepney==0.4.3,joblib==0.15.1,keyring==21.2.1,llvmlite==0.32.1,mccabe==0.6.1,more-itertools==8.3.0,munch==2.5.0,netCDF4==1.5.3,nodeenv==1.3.5,numba==0.49.1,numpy==1.18.4,packaging==20.4,pandas==1.0.3,pathtools==0.1.2,Pint==0.11,pkginfo==1.5.0.1,pluggy==0.13.1,pre-commit==2.4.0,py==1.8.1,pycodestyle==2.6.0,pycparser==2.20,pyflakes==2.2.0,Pygments==2.6.1,pyparsing==2.4.7,pyproj==2.6.1.post1,pytest==5.4.2,pytest-cov==2.9.0,pytest-runner==5.2,python-dateutil==2.8.1,pytz==2020.1,PyYAML==5.3.1,readme-renderer==26.0,requests==2.23.0,requests-toolbelt==0.9.1,Rtree==0.9.4,scikit-learn==0.23.1,scipy==1.4.1,SecretStorage==3.1.2,Shapely==1.7.0,six==1.15.0,threadpoolctl==2.0.0,toml==0.10.1,tox==3.15.1,tqdm==4.46.0,twine==3.1.1,urllib3==1.25.9,virtualenv==20.0.21,watchdog==0.10.2,wcwidth==0.1.9,webencodings==0.5.1,xarray==0.15.1,xclim==0.17.2b0,zipp==3.1.0
py36 run-test-pre: PYTHONHASHSEED='3121258727'
py36 run-test: commands[0] | /home/tjs/git/xclim/.tox/py36/bin/python /snap/pycharm-community/194/plugins/python-ce/helpers/pycharm/_jb_pytest_runner.py --offset 1 -- --cov xclim --basetemp=/home/tjs/git/xclim/.tox/py36/tmp
ImportError while loading conftest '/home/tjs/git/xclim/tests/conftest.py'.
tests/conftest.py:8: in <module>
    import xclim
xclim/__init__.py:5: in <module>
    from xclim.indicators import anuclim
xclim/indicators/__init__.py:255: in <module>
    ICCLIM = __build_icclim("ignore")
xclim/indicators/__init__.py:84: in __build_icclim
    from xclim import indices
xclim/indices/__init__.py:24: in <module>
    from ._anuclim import *
xclim/indices/_anuclim.py:3: in <module>
    from ._multivariate import daily_temperature_range
xclim/indices/_multivariate.py:6: in <module>
    from . import fwi
xclim/indices/fwi.py:85: in <module>
    from dask.array import Array as dskarray
.tox/py36/lib/python3.6/site-packages/dask/array/__init__.py:250: in <module>
    raise ImportError(str(e) + "\n\n" + msg)
E   ImportError: No module named 'tlz'
E
E   Dask array requirements are not installed.
E
E   Please either conda or pip install as follows:
E
E     conda install dask                 # either conda install
E     python -m pip install "dask[array]" --upgrade  # or python -m pip install
Launching pytest with arguments --cov xclim --basetemp=/home/tjs/git/xclim/.tox/py36/tmp in /home/tjs/git/xclim

ERROR: InvocationError for command /home/tjs/git/xclim/.tox/py36/bin/python /snap/pycharm-community/194/plugins/python-ce/helpers/pycharm/_jb_pytest_runner.py --offset 1 -- --cov xclim --basetemp=/home/tjs/git/xclim/.tox/py36/tmp (exited with code 4)

commands failed
___________________________________ summary ____________________________________
ERROR:   py36: commands failed

Process finished with exit code 1

Assertion failed
```

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #465 
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
This PR changes the dask dependency to use `[array]` in lieu of `[complete]`. This is raised due to a bug present in dask@2.17.1 where [complete] was erroneously removed. On examination, the test ensemble doesn't perform `.to_dataframe()` or `distributed` operations, so the reasoning here is that we shouldn't be installing the libraries needed for these operations in a "vanilla" xclim installation.

* **Does this PR introduce a breaking change?** 
No.

* **Other information**:

### CODE REVIEW

**Comment by aulemahal** [comment]
Maybe we should add a note in the docs about this?
Also, I think that distributed is used in the  workflow example notebook. The build may not fail, but that's because notebooks aren't executed.

**Comment by Zeitsperre** [comment]
This is true. Leads me to believe that it might be a good idea to add a new recipe with those libraries installed or to simply document that `distributed` is an optional dependency for doing "cool stuff". I'll add to this PR and ask for another review.

**Comment by huard** [comment]
It could be in setup.py `docs_requirements` only.

File: environment.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ dependencies:
  - shapely>=1.6
  - geopandas>=0.7
  - dask>=2.6.0
+ - toolz>=0.10.0
```
Comment (aulemahal) [inline environment.yml:19]: I believe this [array] / [complete] stuff is only relevant for pip? Conda installs the equivalent of dask[complete], thus I think `toolz` isn't needed as an explicit dependency of xclim!

Comment (Zeitsperre) [reply environment.yml:19]: Understoud! 
```suggestion
```

**Pull request merged**